hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -540,6 +547,51 @@ private HoodieFunctionalIndexDefinition getFunctionalIndexDefinition(String inde
     }
   }
 
+  private Set<String> getSecondaryIndexPartitionsToInit() {
+    Set<String> secondaryIndexPartitions = dataMetaClient.getFunctionalIndexMetadata().get().getIndexDefinitions().values().stream()
+        .map(HoodieFunctionalIndexDefinition::getIndexName)
+        .filter(indexName -> indexName.startsWith(HoodieTableMetadataUtil.PARTITION_NAME_SECONDARY_INDEX_PREFIX))
```

Comment:
So secondary index belongs to functional index?

Reply:
Looks like we have some code that can share with functional index.

Reply:
actually, i should rename `HoodieFunctionalIndexDefinition` to `HoodieIndexDefinition` as it will contain the definition for both functional index and secondary index, and any future record level index.